src/base58.h

```diff
@@ -60,4 +60,15 @@ NODISCARD bool DecodeBase58Check(const char* psz, std::vector<unsigned char>& vc
  */
 NODISCARD bool DecodeBase58Check(const std::string& str, std::vector<unsigned char>& vchRet);
 
+/**
+ * Compute a lower bound on the vchRet result from DecodeBase58Check.
+ *
+ * It will never overshoot, and will never undershoot by more than 0.25% plus 2.
```

Comment:
"and will never undershoot by more than 0.25% plus 2" is not true in the case of trailing spaces, I think :smile:

Reply:
You're right, and that makes this whole approach wrong, as it will reject inputs that are too long just due to spaces at the end. I've changed the approach.